### PR TITLE
48 be implement the blog post tags api endpoint

### LINKED ISSUE #48: [BE] Implement the blog post tags API endpoint
## Business Requirements
As a visitor, I want to be able to view a list of blog post tags that are available so that I can filter blog posts by specific blog post tag.

## Acceptance Criteria
1. There should be an endpoint which allows for viewing a list of available blog post tags, paginated, 20 per page.
3. The blog posts tags should be ordered alphabetically by name.
4. There should be an API documentation generated by Scribe so that the frontend team can easily integrate it.

### PR DESCRIPTION
**Pull request by nikoladomazetovikj**
- Added list blogpost tags API, paginated by 20 and logic for searching by tag name
- Generated `scribe` API docs

### CODE REVIEW

File: app/Website/src/Http/Controllers/BlogTagsController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    #[Endpoint('Website/List Blog Post Tags', <<<'DESC'
+              This endpoint list all blogpost tags in alphabetically order paginated by 20.
+              Additionally you may use ?name= query param to search by name
+    DESC)]
+    #[QueryParam('name', 'string', required: false)]
+    public function index(Request $request): AnonymousResourceCollection
+    {
+        $tags = BlogPostTag::when($request->has('name'), function ($query) use ($request) {
+            return $query->where('name', 'like', "{$request->name}%");
```
Comment (davorminchorov) [inline app/Website/src/Http/Controllers/BlogTagsController.php]: There's no need to filter the tags by any parameter.

File: app/Website/src/Http/Controllers/BlogTagsController.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        })->orderBy('name', 'asc')->paginate(20);
+
+        return BlogPostTagResource::collection($tags);
+    }
+
+    // TODO: In future we may use other methods to create/update tags
```
Comment (davorminchorov) [inline app/Website/src/Http/Controllers/BlogTagsController.php]: You can remove this comment. The tags management will be done in the admin panel.

**Review by davorminchorov — CHANGES_REQUESTED** [review]

File: app/Website/src/Http/Controllers/BlogTagsController.php (hunk 2 above)
Comment (davorminchorov) [reply app/Website/src/Http/Controllers/BlogTagsController.php:29]: It was a copy paste mistake, sorry about that.

File: app/Website/src/Http/Controllers/BlogTagsController.php (hunk 1 above)
Comment (davorminchorov) [reply app/Website/src/Http/Controllers/BlogTagsController.php]: It was a copy paste mistake, sorry about that.

File: app/Website/src/Http/Controllers/BlogTagsController.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class BlogTagsController extends Controller
+{
+    #[Endpoint('Website/List Blog Post Tags', <<<'DESC'
+              This endpoint list all blogpost tags in alphabetically order paginated by 20.
+              Additionally you may use ?name= query param to search by name
+    DESC)]
+    #[QueryParam('name', 'string', required: false)]
+    public function index(Request $request): AnonymousResourceCollection
+    {
+        $tags = BlogPostTag::when($request->has('name'), function ($query) use ($request) {
```
Comment (davorminchorov) [inline app/Website/src/Http/Controllers/BlogTagsController.php]: We won't need to filter tags by anything, the blog posts will be filtered by tag, which is a different issue.
The issue was written with copy paste, I forgot to remove that line from the description.

You can remove the whole when call, and the `name` query parameter.

Comment (nikoladomazetovikj) [reply app/Website/src/Http/Controllers/BlogTagsController.php]: Sorry I forgot to push the commit

**Review by davorminchorov — APPROVED** [review]

**Pull request merged**